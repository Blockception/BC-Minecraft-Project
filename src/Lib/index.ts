/*	Auto generated	*/

export * from "./mcattributes";
export * from "./mcdefinitions";
export * from "./mcignore";
export * from "./project";