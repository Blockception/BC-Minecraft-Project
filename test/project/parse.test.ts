import { TestFilesFolder } from "../utillity";
import * as path from "path";
import { MCProject } from "../../src/main";

describe("MCProject", () => {
  it("loadSync p1", () => {
    const folder = path.join(TestFilesFolder, "mcproject", "p1");
    const project = MCProject.loadSync(folder);
    expect(project).toMatchSnapshot();
  });

  it("load file2", async () => {
    const folder = path.join(TestFilesFolder, "mcproject", "p1");

    const project = await MCProject.load(folder);
    expect(project).toMatchSnapshot();
  });

  it("loadSync not existing", () => {
    const folder = path.join(TestFilesFolder, "mcproject", "non_existing");
    const project = MCProject.loadSync(folder);
    expect(project).toMatchSnapshot();
  });

  it("load not existing", async () => {
    const folder = path.join(TestFilesFolder, "mcproject", "non_existing");

    const project = await MCProject.load(folder);
    expect(project).toMatchSnapshot();
  });
});
