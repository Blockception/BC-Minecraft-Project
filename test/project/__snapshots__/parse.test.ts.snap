// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MCProject load file2 1`] = `
{
  "attributes": {
    "diagnose": "true",
    "diagnose.mcfunctions": "true",
    "diagnose.objectives": "true",
    "diagnose.tags": "false",
    "world.area_used": "0 0 0 1000 256 1000",
  },
  "definitions": {
    "family": {
      "defined": [
        "npc",
        "Npc",
        "enemy",
      ],
      "excluded": [
        "nemesis",
      ],
    },
    "name": {
      "defined": [
        "Steve",
      ],
      "excluded": [
        "Creeper",
      ],
    },
    "objective": {
      "defined": [
        "counter",
      ],
      "excluded": [
        "Foo",
      ],
    },
    "tag": {
      "defined": [
        "tag_number1",
      ],
      "excluded": [
        "FoLolOl",
      ],
    },
  },
  "ignores": {
    "patterns": [
      "OutputFolder",
      "Temp",
      "Template/something",
      "!BP/**/*.json",
    ],
  },
}
`;

exports[`MCProject load not existing 1`] = `
{
  "attributes": {},
  "definitions": {},
  "ignores": {
    "patterns": [],
  },
}
`;

exports[`MCProject loadSync not existing 1`] = `
{
  "attributes": {},
  "definitions": {},
  "ignores": {
    "patterns": [],
  },
}
`;

exports[`MCProject loadSync p1 1`] = `
{
  "attributes": {
    "diagnose": "true",
    "diagnose.mcfunctions": "true",
    "diagnose.objectives": "true",
    "diagnose.tags": "false",
    "world.area_used": "0 0 0 1000 256 1000",
  },
  "definitions": {
    "family": {
      "defined": [
        "npc",
        "Npc",
        "enemy",
      ],
      "excluded": [
        "nemesis",
      ],
    },
    "name": {
      "defined": [
        "Steve",
      ],
      "excluded": [
        "Creeper",
      ],
    },
    "objective": {
      "defined": [
        "counter",
      ],
      "excluded": [
        "Foo",
      ],
    },
    "tag": {
      "defined": [
        "tag_number1",
      ],
      "excluded": [
        "FoLolOl",
      ],
    },
  },
  "ignores": {
    "patterns": [
      "OutputFolder",
      "Temp",
      "Template/something",
      "!BP/**/*.json",
    ],
  },
}
`;
